### PR TITLE
Add `is_deleted` column for project phases and milestones

### PR DESCRIPTION
**Pull request by mddilley**
## Associated issues
https://github.com/cityofaustin/atd-data-tech/issues/9210

## Testing
**URL to test:** 
This PR needs to be tested either locally or on moped-test.

**Steps to test:**
**Project phases**
1. Create a new project and then go to the **Timeline** tab
2. Add:
   - A construction phase with a start date
   - A completion phase with an end date
   - A design phase and makes it the current phase (you'll see the project status badge update to **Design**)
   - Any other phase
3. Go to the **Projects** table view and see the **Construction Start** and **Project Completion** dates populated in your project's row
4. Go to the **Timeline** tab and remove all of the phases
5. You'll see:
   - The project status badge update to **Active**
   - **Construction Start** and **Project Completion** columns no longer have dates for your project in the **Projects** table view

**Project milestones**
1. Add and remove milestones from the new project you made to test project phases
2. You can also create a new project with a template and see that the initial milestones populate

---
#### Ship list
- [x] Code reviewed 
- [ ] Product manager approved
- [ ] Added to [QA test script](https://docs.google.com/spreadsheets/d/1n_O6MLh9cwwPf57HUM394Ea-z9uuoEV1-QW4axNZXLE/edit#) if applicable

### CODE REVIEW

File: moped-database/views/project_list_view.sql [hunk 1]
```diff
@@ -82,6 +82,7 @@ AS WITH project_person_list_lookup AS (
       WHERE true
         AND phases.project_id = mp.project_id 
         AND phases.phase_name = 'construction'::text
+        AND phases.is_deleted = false
```
Comment (mddilley) [inline moped-database/views/project_list_view.sql:85]: These two updates to the view are just to make sure that construction and complete phase dates don't stick around after a soft delete.

File: moped-database/migrations/1655844474965_alter_table_public_moped_proj_phases_add_column_is_deleted/up.sql [hunk 2]
```diff
@@ -0,0 +1,10 @@
+alter table "public"."moped_proj_phases" add column "is_deleted" boolean
```
Comment (mddilley) [inline moped-database/migrations/1655844474965_alter_table_public_moped_proj_phases_add_column_is_deleted/up.sql:1]: I thought it was interesting that the new Hasura engine version didn't capitalize the SQL keywords in auto-generated migrations. 🤷

Comment (chiaberry) [reply moped-database/migrations/1655844474965_alter_table_public_moped_proj_phases_add_column_is_deleted/up.sql:1]: same, I kinda wish they'd autocapitalize.

**Review by chiaberry — APPROVED** [review]

**Review by frankhereford — APPROVED** [review]
Looks great Mike - thank you!

I've tested locally using the instructions in your PR comment - worked perfectly. I've gone through the diff as well, and I think it's 🚢 -shape! 🙏

**Review by johnclary — APPROVED** [review]
looks great; works great. thanks Mike! 🚢 🚢 🚢

**Pull request merged**